### ISSUE TITLE
Visualize Weights

### ISSUE THREAD

**Issue by leonardoaraujosantos** [NVIDIA/DIGITS#505]
Hi guys would be possible to add this layer visualization or weights visualization, like on https://github.com/bruckner/deepViz

**Comment by gheinrich** [comment]
Thanks, are you saying it is difficult to navigate through layers or is this difficult to inspect the weights and activations of a given layer? In case you haven't noticed, you can zoom in/out within visualizations by scrolling up/down with your mouse.

**Comment by leonardoaraujosantos** [comment]
Hi you're saying like checking the weights of for instance Alex net layer 5? Correct me if I'm wrong, but scrolling up/down with the mouse on this seems practical?
![layer5](https://cloud.githubusercontent.com/assets/898383/12302228/165ac9e4-ba1c-11e5-8f61-d01401ea89dc.png)

**Comment by lukeyeager** [comment]
It sounds like you're asking for visualizations of what each neuron has learned, rather than simply the weights of each neuron. It sounds like you're asking for visualizations that require "deconvolution".

I've looked into integrating https://github.com/yosinski/deep-visualization-toolbox, which is similar to DeepViz. Last I checked, there wasn't a deconv layer checked into `BVLC/master` yet (and thus not into `NVIDIA/caffe`, making this impossible for now.

Please correct me if I'm wrong.

**Comment by leonardoaraujosantos** [comment]
Yes exactly they rely on a branch of caffe that has deconvolution, but still seems strange to zoom with the mouse just to check the weights right? That's why I was thinking more on a UI improvement even if there is no deconvolution yet

**Comment by lukeyeager** [comment]
You're right - visualizing `conv5` probably isn't going to be very helpful. But `conv1` can be interesting to look at. Do you have a UI improvement suggestion other than adding deconv visualizations?

Our current visualizations are based off of this Caffe notebook:
https://github.com/BVLC/caffe/blob/master/examples/00-classification.ipynb

**Comment by leonardoaraujosantos** [comment]
Fully agree I was just thinking in something that you click on the weight and it maximise/open in another window

**Comment by lukeyeager** [comment]
Oh I see. Yes, I can see how that would be nice. I don't actually embed all of the data on the page now because for some models (i.e. GoogLeNet) the amount of data would be pretty huge. But it would be nice to be able to view the full image and/or download the data in some easy-to-read format (and you're not the first to request it). I'll put it on the v4.0 TODO list. Thanks for the suggestion!

**Comment by mizadyya** [comment]
@lukeyeager Is there any way to download the images from activation of each layer? How can I save the output of each of convolutional layers, when testing a single image?
 @leonardoaraujosantos Did you find any other way for what you were looking for?
